middle_end/semantics_of_primitives.ml

```diff
@@ -142,6 +142,10 @@ let for_primitive (prim : Lambda.primitive) =
     Misc.fatal_errorf "The primitive %a should have been eliminated by the \
         [Closure_conversion] pass."
       Printlambda.primitive prim
+  | Pload8
+  | Pload16 _
+  | Pload _ ->
+    No_effects, Has_coeffects
```

Comment:
I don't know how this information is used, but could No_effect imply that the code can be executed even if the value is not required (e.g.`if pure_expr then e1; ...` could be turned into `e1` if `e1` is declared to have no effects)?  In which case this seems dangerous, since a load could trigger a memory fault.